static/js/rendered_markdown.js

```diff
@@ -155,13 +155,8 @@ exports.update_elements = (content) => {
             return;
         }
 
-        // Moment throws a large deprecation warning when it has to
-        // fallback to the Date() constructor.  This isn't really a
-        // problem for us except in local echo, as the backend always
-        // uses a format that ensures that is unnecessary.
-        moment.suppressDeprecationWarnings = true;
-        const timestamp = moment(time_str);
-        if (timestamp.isValid()) {
+        const timestamp = parseISO(time_str);
+        if (isValid(timestamp)) {
             const text = $(this).text();
             const rendered_time = timerender.render_markdown_timestamp(timestamp, text);
             $(this).text(rendered_time.text);
```

Comment:
I fixed a `blueslip.error` a few lines below here that incorrectly still referred to `Moment` as the library.